Build failure: gnome-decoder

### Steps To Reproduce

Steps to reproduce the behavior:
1. Trying to install gnome-decoder-0.3.1

### Build log

```
warning: Git tree '/home/lucio/docs/git/flake_inferno' is dirty
error: builder for '/nix/store/hmxgqfivwjsl7sdbkxark7bn4z6miyh4-gnome-decoder-0.3.1.drv' failed with exit code 1;
       last 10 log lines:
       > 5   | ... b"node.id\0" ; pub const PW_KEY_NODE_NAME : & [u8 ; 10usize] = b"node.name\0" ; pub const PW_KEY_NODE_NICK : & [u8 ; 10usize] = b"nod...
       >     |                    --------------------------------------------- similarly named constant `PW_KEY_NODE_NAME` defined here
       >
       > For more information about this error, try `rustc --explain E0425`.
       > error: could not compile `pipewire` due to previous error
       > warning: build failed, waiting for other jobs to finish...
       > [36/37] Generating data/com.belmoussaoui.Decoder.metainfo.xml with a custom command
       > FAILED: src/decoder
       > /nix/store/7w1s50i8yplw2gkpf9hpha6n9vci8g98-coreutils-9.1/bin/env CARGO_HOME=/build/source/build/cargo-home /nix/store/cvj43037d5vaxrzrwn6z8qp5nhd9zcng-cargo-1.67.0/bin/cargo build --manifest-path /build/source/Cargo.toml --target-dir /build/source/build/src --release && cp src/release/decoder src/decoder
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/hmxgqfivwjsl7sdbkxark7bn4z6miyh4-gnome-decoder-0.3.1.drv'.
error: 1 dependencies of derivation '/nix/store/f6ymsvmf084p4ah2w3q4rp16s3lanlx5-gnome-decoder-0.3.1-fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/kag1ylby3bvr8mb2y7ji3spp51qd0s93-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/x3rjbrl34biiza2mddfm47m6a10wbzqv-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/p241kjg2vzkg0v65936hgl3p0vs4z2bd-home-manager-generation.drv' failed to build

```

### Additional context

Trying to install using home-manager and [flake](https://github.com/PmicroTO/flake_inferno).

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
@zendo 
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/wlhhpfxdvxa1mhghvq4gcj78wvxhf1mh-nix-info
copying path '/nix/store/wlhhpfxdvxa1mhghvq4gcj78wvxhf1mh-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.94, NixOS, 23.05 (Stoat), 23.05.20230225.b1f87ca`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.2`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

--- a/pkgs/applications/graphics/gnome-decoder/default.nix
+++ b/pkgs/applications/graphics/gnome-decoder/default.nix
@@ -65,6 +65,12 @@ clangStdenv.mkDerivation rec {
 
   LIBCLANG_PATH = "${libclang.lib}/lib";
 
+  # FIXME: workaround for Pipewire 0.3.64 deprecated API change, remove when fixed upstream
+  # https://gitlab.freedesktop.org/pipewire/pipewire-rs/-/issues/55
+  preBuild = ''
+    export BINDGEN_EXTRA_CLANG_ARGS="$BINDGEN_EXTRA_CLANG_ARGS -DPW_ENABLE_DEPRECATED"
+  '';
+
   meta = with lib; {
     description = "Scan and Generate QR Codes";
     homepage = "https://gitlab.gnome.org/World/decoder";

